common/performance.py

```
@@ -0,0 +1,721 @@
+                    model_class=primary_model,
+                    cache_prefix=f"{primary_model.__name__}_view",
+                    data_builder=view_data_builder,
+                    cache_timeout=getattr(settings, "VIEW_CACHE_TTL", 3600),
+                )
+
+                cache_registry.register(manager_name, cache_manager)
+                self._cache_manager = cache_manager
+            else:
+                self._cache_manager = cache_registry.get(manager_name)
+
+    def get_cache_key(self, action_name: str = "default", **kwargs: Any) -> str:
```

Ruff (0.12.2)

Dynamically typed expressions (typing.Any) are disallowed in `**kwargs`

(ANN401)